lib/libzfs/Makefile.am

```diff
@@ -2,6 +2,22 @@ libzfs_la_CFLAGS  = $(AM_CFLAGS) $(LIBRARY_CFLAGS)
 libzfs_la_CFLAGS += $(LIBCRYPTO_CFLAGS) $(ZLIB_CFLAGS)
 libzfs_la_CFLAGS += -fvisibility=hidden
 
+# Less dire than if you override this in the kernel, but autovectorizing
+# random portions seems like a great way to encounter crashes in places
+# that look, on the surface, like there's no SIMD involved...
+#
+# We would keep -mgeneral-regs-only, but on e.g. x64, things like atof
+# can't be defined without implying using FPU regs, so it's a compile
+# time error.
+NOVECTOR := -fno-tree-vectorize
```

Comment:
Does this fix a real problem or is it just a precaution? I would rather not disable auto-vectorization in userspace since we expect the compiler to attempt these things in userspace, where using SIMD should always be safe.

That said, I know for a fact that the compiler will sometimes vectorize the mixing matrix calculations. I am not sure if its vectorization makes a difference, but I am inclined to let it try.